src/docs/schema.dox

```diff
@@ -66,24 +66,23 @@ struct module to describe the types of columns in a table:
   http://docs.python.org/library/struct
 
 <table>
-@hrow{Format, C Type, Java type, Python type, Notes}
+@hrow{Format, C Type, Python type, Notes}
 @row{\c x, N/A, N/A, N/A, pad byte\, no associated value}
-@row{\c b, \c int8_t, \c byte, \c int, signed byte}
-@row{\c B, \c uint8_t, \c byte, \c int, unsigned byte}
-@row{\c h, \c int16_t, \c short, \c int, signed 16-bit}
-@row{\c H, \c uint16_t, \c short, \c int, unsigned 16-bit}
-@row{\c i, \c int32_t, \c int, \c int, signed 32-bit}
-@row{\c I, \c uint32_t, \c int, \c int, unsigned 32-bit}
-@row{\c l, \c int32_t, \c int, \c int, signed 32-bit}
-@row{\c L, \c uint32_t, \c int, \c int, unsigned 32-bit}
-@row{\c q, \c int64_t, \c long, \c int, signed 64-bit}
-@row{\c Q, \c uint64_t, \c long, \c int, unsigned 64-bit}
-@row{\c r, \c uint64_t, \c long, \c int, record number}
-@row{\c s, \c char[], \c String, \c str, fixed-length string}
-@row{\c S, \c char[], \c String, \c str, NUL-terminated string}
-@row{\c t, \c uint8_t, \c byte, \c int, fixed-length bit field}
-@row{\c u, <code>WT_ITEM *</code>, <code>byte[]</code>, \c str,
-	raw byte array}
+@row{\c b, \c int8_t, \c int, signed byte}
+@row{\c B, \c uint8_t, \c int, unsigned byte}
+@row{\c h, \c int16_t, \c int, signed 16-bit}
+@row{\c H, \c uint16_t, \c int, unsigned 16-bit}
+@row{\c i, \c int32_t, \c int, signed 32-bit}
+@row{\c I, \c uint32_t, \c int, unsigned 32-bit}
+@row{\c l, \c int32_t, \c int, signed 32-bit}
+@row{\c L, \c uint32_t, \c int, unsigned 32-bit}
```

Comment:
For `"l"` in C, I think `long` is appropriate (that is used in `packing_inline.h`, is this the original complaint in the ticket?).  For `"L"` I think `unsigned long` for the C type.

Reply:
I was wrong.  I think you have the documentation correct for this - it matches [Python's struct.pack](https://docs.python.org/3/library/struct.html#format-characters), but I think the code for `WT_PACK_GET` is incorrect:
```
        case 'b':                                                  \
        case 'h':                                                  \
        case 'i':                                                  \
            (pv).u.i = va_arg(ap, int);                            \
            break;                                                 \
        case 'B':                                                  \
        case 'H':                                                  \
        case 'I':                                                  \
        case 't':                                                  \
            (pv).u.u = va_arg(ap, unsigned int);                   \
            break;                                                 \
        case 'l':                                                  \
            (pv).u.i = va_arg(ap, long);                           \
            break;                                                 \
        case 'L':                                                  \
            (pv).u.u = va_arg(ap, unsigned long);                  \
            break;                                                 \
```
Each of these should be written using the types in `WT_UNPACK_PUT`.  Instead of `int` and `long`, it should use `int8_t`, `int16_t`, `int32_t` `int32_t` for `b`,`h`,`i`,`l` respectively at least for the signed types.  Practically speaking, with most compilers and systems, it's not going to matter for `b`, `h`, `i`.  But I think it might for `l` and `L`.  `long` and `unsigned long` are 8 bytes on Ubuntu and OS/X.  If we ask callers to use `uint32_t` for a type, and tell `varargs` that it is a `long`, that could be a problem.

Reply:
We may have been "lucky" up til now.  Anyone that wants a 4 byte value is likely to use `i`/`I`, and for an 8 byte value, they would likely use `q`/`Q`.

Reply:
I tried to come up with a test case where this breaks down, and have failed.  I suspect that in a varargs funbction, compilers tend to upgrade ints to 8 bytes on the stack so that uint32_t, unsigned long behave the same.  And if I make the change that I suggested, I get:
```
../src/include/packing_inline.h:232:35: error: 'int8_t {aka signed char}' is promoted to 'int' when passed through '...' [-Werror]
             (pv).u.i = va_arg(ap, int8_t);                         \
```

Reply:
Thanks for your investigation @ddanderson. I tried the same thing and faced the same integral promotion error.
According to the documentation and the processing in `packing_inline.h`, we might want to modify `l` and `L` cases so they don't use the `long` type to avoid confusion. I will push the change, let me know what you think.

Reply:
See b82fb99